src/main/java/model/GameManager.java

```diff
@@ -0,0 +1,37 @@
+package model;
+
+import model.dto.MoveResult;
+import java.util.List;
+
+public class GameManager {
+
+    private final Referee referee;
+    private final Cars cars;
+
+    private GameManager(final Referee referee, final Cars cars) {
+        this.referee = referee;
+        this.cars = cars;
+    }
+
+    public static GameManager createDefault(final List<String> carNames) {
+        Referee referee = new CompareNumberReferee();
+        return new GameManager(referee, Cars.from(carNames));
+    }
+
+    public List<MoveResult> moveEachCar() {
+        cars.actEachCar(this::moveCarByReferee);
+        return cars.getMoveResult();
+    }
+
+    public List<String> getWinners() {
+        return cars.getWinnerNames();
+    }
+
+    private void moveCarByReferee(final Car car) {
+        if (!referee.isSatisfiedCondition()) {
+            return;
+        }
+
+        car.moveForward();
```

Comment:
```suggestion
        if (referee.isSatisfiedCondition()) {
            car.moveForward();
        }
```
이렇게 긍정문 대신 부정문으로 작성하신 이유가 있을까요?

Reply:
조건에 부합하지 않은 경우를 먼저 리턴시키고 기대되는 조건을 가장 마지막에 위치시키는 Return Early Pattern이 있네요!

사실 저도 이런 패턴이 있는지는 몰랐네요.. 단지 조건이 여러 가지인 경우에는 하나씩 걸러내는 게 더 가독성도 좋고, 깔끔한 코드인 것 같아서 위처럼 쓰고 있었는데 저도 덕분에 배웠네요 😅ㅎㅎ

이 글 읽어보시면 좋을 것 같아요
[https://medium.com/swlh/return-early-pattern-3d18a41bba8](url)